index.d.ts

```diff
@@ -0,0 +1,96 @@
+// TypeScript Version: 3.7
+
+import {Test} from 'unist-util-is'
+import * as unist from 'unist'
+
+type Nullish = undefined | null
+
+declare namespace messageControl {
+  /** A comment marker. */
+  interface Marker<N extends unist.Node> {
+    /** name of marker */
+    name: string
+    /** value after name */
+    attributes: string
+    /** parsed attributes */
+    parameters: Record<string, unknown>
+    /** reference to given node */
+    node: N
+  }
+
+  /** parse a possible comment marker node to a Marker */
+  interface MarkerFn<N extends unist.Node> {
+    (node: N): Marker<N> | Nullish
```

Comment:
I don't think this can be `null` or `undefined`.
It will only be called if there is a value.